scripts/Muon/GUI/Common/plot_widget/plotting_canvas/plot_toolbar.py

```diff
@@ -15,25 +16,79 @@
-                          ('Subplots', 'Edit subplots', 'subplots', 'configure_subplots'),
-                          ('Save', 'Save the figure', 'filesave', 'save_figure'),
+                          ('Subplots', 'Edit subplots', 'mdi.settings', 'configure_subplots'),
+                          ('Save', 'Save the figure', 'mdi.content-save', 'save_figure'),
                           (None, None, None, None),
-                          ('Show/hide legend', 'Toggles the legend on/off', "select", 'toggle_legend'),
+                          ('Show/hide legend', 'Toggles the legend on/off', None, 'toggle_legend'),
                           )
-
+        self.is_major_grid_on = False
+        self.is_minor_grid_on = False
         NavigationToolbar.__init__(self, figure_canvas, parent=parent)
 
+    def _init_toolbar(self):
+        for text, tooltip_text, mdi_icon, callback in self.toolitems:
+            if text is None:
+                self.addSeparator()
+            else:
+
+                if mdi_icon:
+                    a = self.addAction(get_icon(mdi_icon), text, getattr(self, callback))
+                else:
+                    a = self.addAction(text, getattr(self, callback))
+                self._actions[callback] = a
+                if tooltip_text is not None:
+                    a.setToolTip(tooltip_text)
+
+        if self.coordinates:
+            self.locLabel = QtWidgets.QLabel("", self)
+            self.locLabel.setAlignment(QtCore.Qt.AlignRight | QtCore.Qt.AlignTop)
+            self.locLabel.setSizePolicy(QtWidgets.QSizePolicy(QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding,
+                                                                                    QtWidgets.QSizePolicy.Ignored)))
+            labelAction = self.addWidget(self.locLabel)
+            labelAction.setVisible(True)
+
+        # Adjust icon size or they are too small in PyQt5 by default
+        dpi_ratio = QtWidgets.QApplication.instance().desktop().physicalDpiX() / 100
+        self.setIconSize(QtCore.QSize(24 * dpi_ratio, 24 * dpi_ratio))
+
     def toggle_legend(self):
         for ax in self.canvas.figure.get_axes():
             if ax.get_legend() is not None:
                 ax.get_legend().set_visible(not ax.get_legend().get_visible())
         self.canvas.figure.tight_layout()
         self.canvas.draw()
+
+    def show_major_gridlines(self):
+        if self.is_major_grid_on:
+            for ax in self.canvas.figure.get_axes():
+                ax.grid(False)
+            self.is_major_grid_on = False
+        else:
+            for ax in self.canvas.figure.get_axes():
+                ax.grid(True , color='black')
+            self.is_major_grid_on = True
+        self.canvas.draw()
+
+    def show_minor_gridlines(self):
+        if self.is_minor_grid_on:
+            for ax in self.canvas.figure.get_axes():
+                ax.minorticks_off()
+            self.is_major_grid_on = False
```

Comment:
Should this be self.is_minor_grid_on? And likewise for the line below?

Reply:
yes it should , thank you